python/lsst/ts/m2gui/controltab/tab_rigid_body_pos.py

```diff
@@ -33,7 +37,326 @@ class TabRigidBodyPos(TabDefault):
+            self.model.utility_monitor.NUM_DIGIT_AFTER_DECIMAL_DISPLACEMENT
+        )
+
+        position = dict()
+        for axis in self._get_axes():
+            double_spin_box = QDoubleSpinBox()
+            double_spin_box.setDecimals(num_digit_after_decimal)
+            double_spin_box.setSingleStep(get_tol(num_digit_after_decimal))
+
+            if axis.startswith("r"):
+                double_spin_box.setRange(
+                    -self.MAX_ROTATION_IN_URAD, self.MAX_ROTATION_IN_URAD
+                )
+            else:
+                double_spin_box.setRange(
+                    -self.MAX_DISTANCE_IN_MM, self.MAX_DISTANCE_IN_MM
+                )
+
+            position[axis] = double_spin_box
+
+        return position
+
+    def _get_axes(self):
+        """Get the axes of position.
+
+        Returns
+        -------
+        `list`
+            Axes of the position.
+        """
+
+        return ["x", "y", "z", "rx", "ry", "rz"]
+
+    @Slot()
+    def _callback_save_position(self):
+        """Callback of the save-position button. This will save the current
+        position to a file."""
+        self.model.save_position()
+
+    @Slot()
+    def _callback_go_to_home(self):
+        """Callback of the go-to-home button. This will move the M2 to the
+        home position."""
+        self.model.go_to_position(0, 0, 0, 0, 0, 0)
+
+    @Slot()
+    def _callback_set_home(self):
+        """Callback of the set-home button. This will set the new home position
+        of M2."""
+        self.model.set_home()
+
+    @Slot()
+    def _callback_jog(self):
+        """Callback of the jog button. This will move the M2 to the new
+        position based on the relative offsets."""
+
+        current_position = self.model.utility_monitor.position
+        self.model.go_to_position(
+            self._target_position_relative["x"].value() + current_position[0],
+            self._target_position_relative["y"].value() + current_position[1],
+            self._target_position_relative["z"].value() + current_position[2],
+            self._target_position_relative["rx"].value() + current_position[3],
+            self._target_position_relative["ry"].value() + current_position[4],
+            self._target_position_relative["rz"].value() + current_position[5],
+        )
+
+    @Slot()
+    def _callback_clear_values(self, target_positions):
+        """Callback of the clear-values button. This will clear the values of
+        new target positions.
+
+        Parameters
+        ----------
+        target_positions : `dict`
+            Target positions of the rigid body. The key is the axis. The value
+            is the `PySide2.QtWidgets.QDoubleSpinBox`.
+        """
+
+        for target_position in target_positions.values():
+            target_position.setValue(0)
+
+    @Slot()
+    def _callback_go_to_position(self):
+        """Callback of the go-to-position button. This will move the M2 to the
```

Comment:
The GUI shall produce warning to operator if this fails. The best way to report failures is to raise exception. See comments in model.go_to_position and add try: .. except block there. See SALComm for an example:

```python
def warning(parent, title, description):
    """Creates future with QMessageBox. Enables use of QMessageBox with
    asyncqt/asyncio. Mimics QMessageBox.warning behaviour - but QMessageBox
    cannot be used, as it blocks Qt loops from executing (as all modal dialogs
    does).

    Parameters
    ----------
    parent : `QtWidget`
        Parent widget.
    title : `str`
        Message window title.
    description : `str`
        Descrption of warning occured.
    """

    future = asyncio.Future()
    dialog = QMessageBox(parent)
    dialog.setWindowTitle(title)
    dialog.setText(description)
    dialog.setIcon(QMessageBox.Warning)
    dialog.finished.connect(lambda r: future.set_result(r))
    dialog.open()
    return future
```

at this stage, you don't need to mess with future, as you don't have async calls (yet?), so creating dialog and executing it would be enough.

Reply:
The error handling will be implemented in DM-34861. Thanks!